Travis build failing, mingw related

> ```
In file included from /home/travis/build/LMMS/lmms/src/core/ConfigManager.cpp:245:0:
/opt/mingw32/i686-w64-mingw32/include/shlobj.h:37:3: error: 'EXTERN_C' does not name a type
   SHFOLDERAPI SHGetFolderPathW (HWND hwnd, int csidl, HANDLE hToken, DWORD dwFlags, LPWSTR pszPath);
```

In this code block:

https://github.com/LMMS/lmms/blob/master/src/core/ConfigManager.cpp#L245

Related commit (different project):

https://github.com/Nuand/bladeRF/commit/c631100589d86db708757d51f004cb36c06c5ec3

@Wallacoloo, thx, it looks like we have a new one now per [`jobs/58370993`](https://travis-ci.org/LMMS/lmms/jobs/58370993)... :)

```
/opt/mingw32/i686-w64-mingw32/include/shlobj.h:37:3: error: 'DECLSPEC_IMPORT' does not name a type
   SHFOLDERAPI SHGetFolderPathW (HWND hwnd, int csidl, HANDLE hToken, DWORD dwFlags, LPWSTR pszPath);
   ^
In file included from /opt/mingw32/i686-w64-mingw32/include/windows.h:108:0,
                 from /opt/mingw32/i686-w64-mingw32/include/rpc.h:16,
                 from /opt/mingw32/i686-w64-mingw32/include/objbase.h:7,
                 from /opt/mingw32/i686-w64-mingw32/include/ole2.h:17,
                 from /opt/mingw32/i686-w64-mingw32/include/shlobj.h:107,
                 from /home/travis/build/LMMS/lmms/src/core/ConfigManager.cpp:258:
/opt/mingw32/i686-w64-mingw32/include/commdlg.h:575:53: error: expected class-name before '{' token
   DECLARE_INTERFACE_(IPrintDialogCallback,IUnknown) {
                                                     ^
/opt/mingw32/i686-w64-mingw32/include/commdlg.h:588:53: error: expected class-name before '{' token
   DECLARE_INTERFACE_(IPrintDialogServices,IUnknown) {
                                                     ^
/opt/mingw32/i686-w64-mingw32/include/commdlg.h:621:5: error: 'LPUNKNOWN' does not name a type
     LPUNKNOWN lpCallback;
     ^
/opt/mingw32/i686-w64-mingw32/include/commdlg.h:645:5: error: 'LPUNKNOWN' does not name a type
     LPUNKNOWN lpCallback;
     ^
```

So what caused the break? Can we trace it back to a specific commit, or was
it a change in the mingw version that Travis tests against? It seems odd
for this problem to just appear from nowhere.
On Apr 13, 2015 6:23 PM, "Tres Finocchiaro" <notifications@github.com>
wrote:

> @Wallacoloo <https://github.com/Wallacoloo>, thx, it looks like we have a
> new one now per jobs/58370993
> <https://travis-ci.org/LMMS/lmms/jobs/58370993>... :)
>
> /opt/mingw32/i686-w64-mingw32/include/shlobj.h:37:3: error: 'DECLSPEC_IMPORT' does not name a type
>    SHFOLDERAPI SHGetFolderPathW (HWND hwnd, int csidl, HANDLE hToken, DWORD dwFlags, LPWSTR pszPath);
>    ^
> In file included from /opt/mingw32/i686-w64-mingw32/include/windows.h:108:0,
>                  from /opt/mingw32/i686-w64-mingw32/include/rpc.h:16,
>                  from /opt/mingw32/i686-w64-mingw32/include/objbase.h:7,
>                  from /opt/mingw32/i686-w64-mingw32/include/ole2.h:17,
>                  from /opt/mingw32/i686-w64-mingw32/include/shlobj.h:107,
>                  from /home/travis/build/LMMS/lmms/src/core/ConfigManager.cpp:258:
> /opt/mingw32/i686-w64-mingw32/include/commdlg.h:575:53: error: expected class-name before '{' token
>    DECLARE_INTERFACE_(IPrintDialogCallback,IUnknown) {
>                                                      ^
> /opt/mingw32/i686-w64-mingw32/include/commdlg.h:588:53: error: expected class-name before '{' token
>    DECLARE_INTERFACE_(IPrintDialogServices,IUnknown) {
>                                                      ^
> /opt/mingw32/i686-w64-mingw32/include/commdlg.h:621:5: error: 'LPUNKNOWN' does not name a type
>      LPUNKNOWN lpCallback;
>      ^
> /opt/mingw32/i686-w64-mingw32/include/commdlg.h:645:5: error: 'LPUNKNOWN' does not name a type
>      LPUNKNOWN lpCallback;
>      ^
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LMMS/lmms/issues/1952#issuecomment-92553119>.
>

No dice... 

Just discovered that CMake is having issues as well per:

```
/opt/mingw32/lib/libogg.dll.a;/opt/mingw32/lib/libvorbis.dll.a;/opt/mingw32/lib/libvorbisfile.dll.a;/opt/mingw32/lib/libvorbisenc.dll.a
CMake Error at /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:306 (message):
  pkg-config tool not found
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:320 (_pkg_check_modules_internal)
  CMakeLists.txt:303 (PKG_CHECK_MODULES)
-- Could NOT find Wine (missing:  WINE_LIBRARIES WINE_INCLUDE_DIRS) 
CMake Error at /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:306 (message):
  pkg-config tool not found
Call Stack (most recent call first):
  /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:320 (_pkg_check_modules_internal)
  CMakeLists.txt:351 (PKG_CHECK_MODULES)
```

I'll go back on a few Travis builds and see if this is normal or not...

That Cmake issue looks to be because Travis doesn't have pkg-config (which
is an application wholly separate from CMake) installed. Perhaps try adding
`apt-get install pkg-config' (or whatever variation is needed based on
permissions) somewhere in travis.yml before the CMake build gets invoked.
Make sure that the pkg-config installation directory is added to the $PATH
if it isn't already.
On Apr 15, 2015 9:34 PM, "Tres Finocchiaro" <notifications@github.com>
wrote:

> No dice...
>
> Just discovered that CMake is having issues as well per:
>
> /opt/mingw32/lib/libogg.dll.a;/opt/mingw32/lib/libvorbis.dll.a;/opt/mingw32/lib/libvorbisfile.dll.a;/opt/mingw32/lib/libvorbisenc.dll.a
> CMake Error at /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:306 (message):
>   pkg-config tool not found
> Call Stack (most recent call first):
>   /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:320 (_pkg_check_modules_internal)
>   CMakeLists.txt:303 (PKG_CHECK_MODULES)
> -- Could NOT find Wine (missing:  WINE_LIBRARIES WINE_INCLUDE_DIRS)
> CMake Error at /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:306 (message):
>   pkg-config tool not found
> Call Stack (most recent call first):
>   /usr/share/cmake-2.8/Modules/FindPkgConfig.cmake:320 (_pkg_check_modules_internal)
>   CMakeLists.txt:351 (PKG_CHECK_MODULES)
>
> I'll go back on a few Travis builds and see if this is normal or not...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LMMS/lmms/issues/1952#issuecomment-93637543>.
>

--- a/src/core/ConfigManager.cpp
+++ b/src/core/ConfigManager.cpp
@@ -252,31 +252,6 @@ void ConfigManager::setValue( const QString & _class,
 
 
 
-#ifdef LMMS_BUILD_WIN32
-#include <QLibrary>
-#include <shlobj.h>
-
-// taken from qt-win-opensource-src-4.2.2/src/corelib/io/qsettings.cpp
-static QString windowsConfigPath( int _type )
-{
-	QString result;
-
-	QLibrary library( "shell32" );
-	typedef BOOL( WINAPI* GetSpecialFolderPath )( HWND, char *, int, BOOL );
-	GetSpecialFolderPath SHGetSpecialFolderPath = (GetSpecialFolderPath)
-	library.resolve( "SHGetSpecialFolderPathA" );
-	if( SHGetSpecialFolderPath )
-	{
-		char path[MAX_PATH];
-		SHGetSpecialFolderPath( 0, path, _type, false );
-		result = QString::fromLocal8Bit( path );
-	}
-	return result;
-}
-#endif
-
-
-
 void ConfigManager::loadConfigFile()
 {
 	// read the XML file and create DOM tree
@@ -381,8 +356,8 @@ void ConfigManager::loadConfigFile()
 			!QDir( m_vstDir ).exists() )
 	{
 #ifdef LMMS_BUILD_WIN32
-		m_vstDir = windowsConfigPath( CSIDL_PROGRAM_FILES ) +
-											QDir::separator() + "VstPlugins";
+		QString programFiles = getenv("ProgramFiles");
+		m_vstDir =  programFiles + QDir::separator() + "VstPlugins";
 #else
 		m_vstDir =  m_workingDir + "plugins/vst" + QDir::separator();
 #endif

